crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -368,10 +368,18 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
             pat.len(), ca.len(),
         );
 
-        let op = |opt_s: Option<&str>, opt_pat: Option<&str>| -> PolarsResult<Option<u32>> {
+        // Very simple cache: don't recompile the same regex for multiple values in a row.
+        // TODO: proper LRU cache with reasonable memory limit.
+        let mut reg = Regex::new("").unwrap();
+        let mut last_pat = String::new();
+
+        let op = move |opt_s: Option<&str>, opt_pat: Option<&str>| -> PolarsResult<Option<u32>> {
             match (opt_s, opt_pat) {
                 (Some(s), Some(pat)) => {
-                    let reg = Regex::new(pat)?;
+                    if pat != last_pat {
+                        reg = Regex::new(pat)?;
+                        last_pat = pat.to_owned();
```

Comment:
We really don't want to do a heap allocation here.

Reply:
@ritchie46 I think it's fine, its cost is dwarfed by the cost to build a regex anyway.

Reply:
On a second thought, `Regex` already supports `.as_str()` which gives you back the original string. So we can simply do `if pat != reg.as_str()` and eliminate `last_pat` entirely.

Reply:
Ah.. good catch.

Reply:
I should've caught this. Much better